File: clients/js/src/features/wrap_feature_interface_invoke.ts
import { PolywrapClientConfigBuilder, PolywrapClient } from "@polywrap/client-js";
import path from "path";

export async function runTestCase(_: unknown): Promise<void> {
  const root = path.join(__dirname, "../../../../wraps");
  const interfaceUri = "wrap://authority/interface";
  const implementationPath = path.join(root, "/interface-invoke/01-implementation/implementations/as");
  const implementationUri = `fs/${implementationPath}`;

  const config = new PolywrapClientConfigBuilder()
    .addDefaults()
    .addInterfaceImplementation(interfaceUri, implementationUri);

  const client = new PolywrapClient(config.build());

  const wrapperPath = path.join(root, "/interface-invoke/02-wrapper/implementations/as");
  const wrapperUri = `fs/${wrapperPath}`;

  console.log("Invoking moduleMethod");

  const result = await client.invoke({
    uri: wrapperUri,
    method: "moduleMethod",
    args: {
      arg: {
        uint8: 1,
        str: "Test String 1",
      },
    },
  });

  if (!result.ok) throw result.error;

  console.log("Success!");
}

File: clients/py/src/features/wrap_feature_interface_invoke.py
from pathlib import Path
from typing import Any

from polywrap import (
    FsUriResolver,
    PolywrapClient,
    PolywrapClientConfigBuilder,
    SimpleFileReader,
    Uri,
)


def run_test_case(_: Any) -> None:
    root = Path(__file__).parent.parent.parent.parent.parent / "wraps"
    interface_uri = Uri.from_str("wrap://authority/interface")
    implementation_path = root / "interface-invoke/01-implementation/implementations/as"
    implementation_uri = Uri.from_str(f"fs/{implementation_path}")

    config = (
        PolywrapClientConfigBuilder()
        .add_interface_implementations(interface_uri, [implementation_uri])
        .add_resolver(FsUriResolver(SimpleFileReader()))
        .build()
    )

    client = PolywrapClient(config)

    wrapper_path = root / "interface-invoke/02-wrapper/implementations/as"
    wrapper_uri = Uri.from_str(f"fs/{wrapper_path}")

    print("Invoking moduleMethod")

    if result := client.invoke(
        uri=wrapper_uri,
        method="moduleMethod",
        args={
            "arg": {
                "uint8": 1,
                "str": "Test String 1",
            },
        },
    ):
        print("Success!")
    else:
        raise ValueError(f"Error: {result}")
